packages/oracle/src/dialect.ts

```diff
@@ -0,0 +1,215 @@
+  zerofill: false,
+  unsigned: true,
+};
+
+export class OracleDialect extends AbstractDialect<OracleDialectOptions, OracleConnectionOptions> {
+  static readonly supports = AbstractDialect.extendSupport({
+    'VALUES ()': true,
+    'LIMIT ON UPDATE': true,
+    lock: false,
+    forShare: 'LOCK IN SHARE MODE',
+    index: {
+      collate: false,
+      length: false,
+      parser: false,
+      type: false,
+      using: false,
+    },
+    constraints: {
+      restrict: false,
+      onUpdate: false,
+    },
+    returnValues: false,
+    returnIntoValues: true,
+    'ORDER NULLS': true,
+    schemas: true,
+    inserts: {
+      updateOnDuplicate: false,
+    },
+    indexViaAlter: false,
+    dataTypes: {
+      COLLATE_BINARY: true,
+      GEOMETRY: false,
+      JSON: true,
+      INTS: numericOptions,
+      DOUBLE: numericOptions,
+      DECIMAL: { unconstrained: true },
+      TIME: {
+        precision: false,
+      },
+    },
+    jsonOperations: true,
+    jsonExtraction: {
+      quoted: true,
+    },
+    dropTable: {
+      cascade: true,
+    },
+    renameTable: {
+      changeSchema: false,
+    },
+    delete: {
+      limit: true,
+    },
+    startTransaction: {
+      useBegin: true,
+    },
+    upserts: true,
+    bulkDefault: true,
+    topLevelOrderByRequired: true,
+  });
+
+  readonly connectionManager: OracleConnectionManager;
+  readonly queryGenerator: OracleQueryGenerator;
+  readonly queryInterface: OracleQueryInterface;
+  readonly Query = OracleQuery;
+  readonly dataTypesDocumentationUrl =
+    'https://docs.oracle.com/en/database/oracle/oracle-database/23/sqlrf/Data-Types.html';
+
+  constructor(sequelize: Sequelize, options: OracleDialectOptions) {
+    super({
+      dataTypesDocumentationUrl:
+        'https://docs.oracle.com/en/database/oracle/oracle-database/23/sqlrf/Data-Types.html',
+      minimumDatabaseVersion: '18.0.0',
+      identifierDelimiter: '"',
+      name: 'oracle',
+      options,
+      sequelize,
+      dataTypeOverrides: DataTypes,
+    });
+
+    this.connectionManager = new OracleConnectionManager(this);
+    this.queryGenerator = new OracleQueryGenerator(this);
+    this.queryInterface = new OracleQueryInterface(this);
+  }
+
+  parseConnectionUrl(url: string): OracleConnectionOptions {
+    return parseCommonConnectionUrlOptions<OracleConnectionOptions>({
+      url,
+      allowedProtocols: ['oracle'],
+      hostname: 'host',
+      port: 'port',
+      pathname: 'database',
+      username: 'user',
+      password: 'password',
+      stringSearchParams: STRING_CONNECTION_OPTION_NAMES,
+      booleanSearchParams: BOOLEAN_CONNECTION_OPTION_NAMES,
+      numberSearchParams: NUMBER_CONNECTION_OPTION_NAMES,
+    });
+  }
+
+  getDefaultSchema(): string {
+    return this.sequelize.options.replication.write.username?.toUpperCase() ?? '';
+  }
+
+  createBindCollector() {
+    return createNamedParamBindCollector(':');
+  }
+
+  static getDefaultPort(): number {
```

Comment:
does not exist anymore because connection options are now fully dialect-specific

Reply:
Removed.